src/pages/Home/components/Portfolio/Prompt/Prompt.tsx

```diff
@@ -0,0 +1,459 @@
+        addressBitcoin?: string;
+        addressAvalanche?: string;
+      }) => {
+        await createContact({
+          id: '',
+          name,
+          address,
+          addressBTC: addressBitcoin,
+          addressXP: addressAvalanche,
+        });
+
+        return {
+          content: `Success! New contact added!`,
+        };
+      },
+      swap: async ({
+        amount,
+        fromTokenAddress,
+        toTokenAddress,
+      }: {
+        amount: number;
+        fromTokenAddress: string;
+        toTokenAddress: string;
+      }) => {
+        const srcToken = tokens.find(
+          (item) =>
+            item.symbol === fromTokenAddress ||
+            ('address' in item && item.address === fromTokenAddress),
+        );
+        const toToken = tokens.find(
+          (item) =>
+            item.symbol === toTokenAddress ||
+            ('address' in item && item.address === toTokenAddress),
+        );
+        if (
+          !srcToken ||
+          (srcToken.type !== TokenType.ERC20 &&
+            srcToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the source token`);
+        }
+        if (
+          !toToken ||
+          (toToken.type !== TokenType.ERC20 &&
+            toToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the destination token`);
+        }
+
+        const amountBigInt = stringToBigint(
+          amount.toString(),
+          srcToken.decimals,
+        );
+
+        const rate = await getRate({
+          srcDecimals: srcToken.decimals,
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          destDecimals: toToken.decimals,
+          srcAmount: amountBigInt.toString(),
+        });
+        if (
+          isAPIError(rate.optimalRate) ||
+          !rate.optimalRate ||
+          !rate.destAmount
+        ) {
+          throw new Error(
+            `Error while getting swap rate. ${isAPIError(rate.optimalRate) ? rate.optimalRate.message : ''}`,
+          );
+        }
+
+        const result = await swap({
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          srcDecimals: srcToken.decimals,
+          destDecimals: toToken.decimals,
+          srcAmount: rate.optimalRate.srcAmount,
+          priceRoute: rate.optimalRate,
+          destAmount: rate.destAmount,
+          gasLimit: Number(rate.optimalRate.gasCost),
+          slippage: 0.15,
+        });
+
+        return {
+          content: `Swap successful. Successfully swapped ${amount}${srcToken.symbol} to ${rate.destAmount}${toToken.symbol}. Tx hash: ${result.swapTxHash}`,
```

Comment:
Message is misleading. `swap()` going through only means that transaction was submitted, it doesn't necessarily mean it was successful (could be reverted, get stuck in mempool, etc.).